lib/preview/image.js

```diff
@@ -236,18 +241,41 @@ var cropImage = function(ctx, file, options, callback) {
  * @api private
  */
 var resizeImage = function(ctx, file, options, callback) {
-  var outputFilename = util.format('resized_%dx%d_%d.png', options.width, options.height, Date.now());
-  var outputFilePath = path.join(ctx.directory, outputFilename);
+  var gmChain = gm(file);
+  
+  gmChain.identify(function(err, metadata) {
+    if (err) {
+      return callback({'code': 500, 'msg': err.message});
+    }
 
-  gm(file)
-    .autoOrient()
-    .noProfile()
-    .resize(options.width, options.height)
-    .write(outputFilePath, function(err) {
-      if (err) {
-        return callback({'code': 500, 'msg': err.message});
+    // Most files will be resized as PNG.
+    var extension = 'png';
+
+    // However, the presence of Scene metadata indicates an animated GIF.
+    if (metadata.format === 'GIF' && metadata.Scene) {
+      if (options.width <= PreviewConstants.SIZES.IMAGE) {
+        // If an animated GIF is below our size threshold, don't process; just return the original file.
+        return callback(null, file);
+      } else {
+        // Otherwise, resize but preserve the GIF format.
+        extension = 'gif';
       }
+    }
 
-      return callback(null, outputFilePath);
-    });
+    var outputFilename = util.format('resized_%dx%d_%d.%s', options.width, options.height, Date.now(), extension);
+    var outputFilePath = path.join(ctx.directory, outputFilename);
+
+    gmChain
+      .autoOrient()
+      .noProfile()
+      .coalesce()
```

Comment:
The `coalesce` option helps prevent artifacts in resized animations. http://www.imagemagick.org/Usage/anim_basics/#coalesce